lib/packwerk/parse_run.rb

```diff
@@ -76,16 +94,25 @@ def check
 
     private
 
-    sig { params(run_context: RunContext, show_errors: T::Boolean).returns(OffenseCollection) }
-    def find_offenses(run_context, show_errors: false)
-      offense_collection = OffenseCollection.new(@configuration.root_path)
+    sig do
+      params(
+        run_context: RunContext,
+        block: T.nilable(T.proc.params(
+          offenses: T::Array[Packwerk::Offense],
+        ).void)
+      ).returns(T::Array[Offense])
+    end
+    def find_offenses(run_context, &block)
       all_offenses = T.let([], T::Array[Offense])
-      process_file = T.let(->(relative_file) do
-        run_context.process_file(relative_file: relative_file).tap do |offenses|
-          failed = show_errors && offenses.any? { |offense| !offense_collection.listed?(offense) }
-          update_progress(failed: failed)
-        end
-      end, ProcessFileProc)
+      process_file = if block
+        T.let(proc do |relative_file|
+          run_context.process_file(relative_file: relative_file).tap(&block)
+        end, ProcessFileProc)
+      else
+        T.let(proc do |relative_file|
+          run_context.process_file(relative_file: relative_file)
+        end, ProcessFileProc)
+      end
```

Comment:
Can we move this to a private method? This is slightly repetitive, but I think it is justified with how many times this gets  called.

Reply:
Do you mean something like this?

```rb
def process_file_proc(&block)
  if block
    T.let(proc do |relative_file|
      run_context.process_file(relative_file: relative_file).tap(&block)
    end, ProcessFileProc)
  else
    T.let(proc do |relative_file|
      run_context.process_file(relative_file: relative_file)
    end, ProcessFileProc)
  end
end

# Then I can call it where it's used:
# ...
      @progress_formatter.started_inspection(@relative_file_set) do
        all_offenses = if @configuration.parallel?
          Parallel.flat_map(@relative_file_set, &process_file_proc(&block))
        else
          serial_find_offenses(&process_file_proc(&block))
        end
      end
# ...
# (Or I can keep a local variable if that seems better.)
```